kytos/core/user.py

```diff
@@ -0,0 +1,93 @@
+"""User authentification """
+import hashlib
+from datetime import datetime
+from typing import Literal, Optional
+
+# pylint: disable=no-name-in-module
+from pydantic import BaseModel, EmailStr, Field, constr, validator
+
+
+class DocumentBaseModel(BaseModel):
+    """DocumentBaseModel"""
+
+    id: str = Field(None, alias="_id")
+    inserted_at: Optional[datetime]
+    updated_at: Optional[datetime]
+    deleted_at: Optional[datetime]
+
+    def dict(self, **kwargs) -> dict:
+        """Model to dict."""
+        values = super().dict(**kwargs)
+        if "id" in values and values["id"]:
+            values["_id"] = values["id"]
+        if "exclude" in kwargs and "_id" in kwargs["exclude"]:
+            values.pop("_id")
+        return values
+
+
+class UserDoc(DocumentBaseModel):
+    """UserDocumentModel."""
+
+    username: constr(min_length=1)
```

Comment:
@Alopalao let's take the opportunity to also set a max_length of 64, since this is exposed in the API, we also don't want to allow massive strings to be stored. 

I'm also thinking that we should probably only allow `[a-z|A-Z|0-9|_|-]` ASCII alphabet, numbers, underscore and hyphen on username for now, and then if we need to support other chars or unicode in the future we can, currently, username could have spaces and or unicode symbols, this probably won't lead to issues in the backend side, but we can very odd usernames, and since they're used in URL routes, some clients might have encoding issues (or at least some initial encoding difficulties to sort it out). So, let's start it stricter and if needed we loosen the username constraints. Let me know if you have any suggestion about this.

Reply:
Is it okay if I use regex?

Reply:
@Alopalao, yes, that'll probably be the easiest to maintain and evolve. If we manage to have great test cov for it, that'd be great.

Reply:
The commit looks great, Aldo, very well integrated with `pydantic` validation model. If you could add some unit tests to cover it, that'd be great, thanks.

Reply:
Great, working on it.